samples/excel/85-preview-apis/chart-data-source.yaml

```
@@ -0,0 +1,129 @@
+order: 6
+id: excel-chart-data-source
+name: Chart series data source
+description: This sample shows how to get information about the data source of a chart series.
+host: EXCEL
+api_set:
+    ExcelApi: BETA (PREVIEW ONLY)
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
+        $("#log-chart-series-source").click(() => tryCatch(logChartSeriesSource));
+
+        async function logChartSeriesSource() {
+          await Excel.run(async (context) => {
+            // This method retrieves the data source string and type values from
+            // the chart on the Sample worksheet.
+            const sheet = context.workbook.worksheets.getItem("Sample");
+
+            // Get the series collection from the first chart on the worksheet, and
+            // then get the first series in that collection.
+            const seriesCollection = sheet.charts.getItemAt(0).series;
+            const series = seriesCollection.getItemAt(0);
+
+            // Get the series data source string and type values. 
+            const dataSourceString = series.getDimensionDataSourceString("Values");
+            const dataSourceType = series.getDimensionDataSourceType("Values");
+            await context.sync();
+            
+            // Log the data source information to the console.
+            console.log("Qtr1 data source string: " + dataSourceString.value);
+            console.log("Qtr1 data source type: " + dataSourceType.value);
+          });
+        }
+
+        async function setup() {
+          await Excel.run(async (context) => {
+            // Create a new worksheet called "Sample" and activate it.
+            context.workbook.worksheets.getItemOrNullObject("Sample").delete();
+            const sheet = context.workbook.worksheets.add("Sample");
+            
+            // Create an a table named SalesTable on the Sample worksheet.
+            let expensesTable = sheet.tables.add("A1:E1", true);
+            expensesTable.name = "SalesTable";
+
+            expensesTable.getHeaderRowRange().values = [["Product", "Qtr1", "Qtr2", "Qtr3", "Qtr4"]];        
+            expensesTable.rows.add(null, [
+              ["Frames", 5000, 7000, 6544, 4377],
+              ["Saddles", 400, 323, 276, 651],
+              ["Brake levers", 12000, 8766, 8456, 9812],
+              ["Chains", 1550, 1088, 692, 853],
+              ["Mirrors", 225, 600, 923, 544],
+              ["Spokes", 6005, 7634, 4589, 8765]
+            ]);
+            
+            sheet.getUsedRange().format.autofitColumns();
+            sheet.getUsedRange().format.autofitRows();
+            sheet.activate();
+
+            // Create a line chart based on data from Qtr1 from SalesTable.
+            let dataRange = sheet.getRange("A1:B7");
+            let chart = sheet.charts.add("Line", dataRange, "Auto");
```

Consider having the chart show part sales over time, instead of having each series be a quarter. This would have the following effects:
- The scenario is easier to logically follow.
- The chart will have multiple series, so it highlights the specifics of what the new API returns.
- It gives you some code to copy and paste as a sample for `ChartSeriesBy`.

```suggestion
            // Create a line chart based on data from SalesTable.
            let dataRange = sheet.getRange("A1:E7");
            let chart = sheet.charts.add("Line", dataRange, Excel.ChartSeriesBy.rows);
```